src/cpp/server/DBActiveSessionStorage.cpp

```diff
@@ -0,0 +1,404 @@
+{
+   for(size_t i=0; i < iter->size(); i++)
+   {
+      std::string key = iter->get_properties(i).get_name();
+      if(key != kUserId)
+      {
+         pTargetMap->insert(
+            std::pair<std::string, std::string>{
+               key,
+               iter->get<std::string>(key, "")
+            }
+         );
+      }
+      else
+      {
+         pTargetMap->insert(
+            std::pair<std::string, std::string>{
+               key,
+               std::to_string(iter->get<int>(key))
+            }
+         );
+      }
+   }
+}
+} // anonymous namespace
+
+Error getConn(boost::shared_ptr<database::IConnection>* connection) {
+   bool success = server_core::database::getConnection(boost::posix_time::milliseconds(500), connection);
+
+   if(!success)
+   {
+      return Error{"FailedToAcquireConnection", errc::ConnectionFailed, "Failed to acquire a connection in 500 milliseconds.", ERROR_LOCATION};
+   }
+
+   return Success();
+}
+
+Error DBActiveSessionStorage::getConnectionOrOverride(boost::shared_ptr<database::IConnection>* connection)
+{
+   if(overrideConnection_ == nullptr)
+      return getConn(connection);
+   else
+   {
+      *connection = overrideConnection_;
+      return Success();
+   }
+}
+
+DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId)
+   : sessionId_(sessionId)
+{
+   
+}
+
+DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId, boost::shared_ptr<core::database::IConnection> overrideConnection)
+   : sessionId_(sessionId), overrideConnection_(overrideConnection)
+{
+
+}
+
+Error DBActiveSessionStorage::readProperty(const std::string& name, std::string* pValue)
+{
+   *pValue = "";
+   boost::shared_ptr<database::IConnection> connection;
+   Error error = getConnectionOrOverride(&connection);
+   
+   if(error)
+   {
+      return error;
+   }
+
+
+   database::Query query = connection->query("SELECT "+name+" FROM "+kTableName+" WHERE "+kSessionIdColumnName+" = :id")
+      .withInput(sessionId_);
+
+   database::Rowset results{};
+   error = connection->execute(query, results);
+
+   if(!error)
+   {
+      database::RowsetIterator iter = results.begin();
+      if(iter != results.end())
+      {
+         if(name != kUserId)
+         {
+            *pValue = iter->get<std::string>(name, "");
+         }
+         else
+         {
+            *pValue = std::to_string(iter->get<int>(name));
+         }
+         if(++iter != results.end()) 
+         {
+            return Error{"Too many sessions returned", errc::TooManySessionsReturned, ERROR_LOCATION};
+         }
+      }
+      else
+      {
+         return Error{"Session does not exist", errc::SessionNotFound, ERROR_LOCATION};
+      }
+   }
+   else
+   {
+      return Error{"DatabaseException", errc::DBError, "Database exception during property read [ session:"+sessionId_+" property:"+name+" ]", error, ERROR_LOCATION};
+   }
```

Comment:
Nit: re-org for less indentation
```suggestion
   error = connection->execute(query, results);
   if (error)
      return Error("DatabaseException", errc::DBError, "Database exception during property read [ session:"+sessionId_+" property:"+name+" ]", error, ERROR_LOCATION);

   database::RowsetIterator iter = results.begin();
   if(iter != results.end())
   {
      if(name != kUserId)
         *pValue = iter->get<std::string>(name, "");
      else
         *pValue = std::to_string(iter->get<int>(name));

      if(++iter != results.end()) 
         return Error("Too many sessions returned", errc::TooManySessionsReturned, ERROR_LOCATION);
   }
   else
      return Error("Session does not exist", errc::SessionNotFound, ERROR_LOCATION);
```

Reply:
Can you make similar adjustments throughout?